### ISSUE TITLE
Ruffle overwrites Flash-era YouTube embeds even though they load the HTML5 player

### ISSUE THREAD

**Issue by WumboSpasm** [ruffle-rs/ruffle#4030]
**Describe the bug**
Old YouTube embeds that were meant to load the Flash player are being replaced by the Ruffle canvas (which obviously doesn't work). However, Google has put in place code that overwrites these embeds with an iframe containing the modern player, meaning there is no reason for Ruffle to intervene.

Examples:
[Case 1](http://www.gershonkingsley.com/popcorn.html)
[Case 2](https://answers.microsoft.com/en-us/xbox/forum/xba_wowpc-xba_gowcc/why-is-windows-10-not-installed-on-xbox-series-x/98498b7d-8461-40f2-8380-1cc1d1622fe9)

**Expected behavior**
The modern HTML5 player should be loaded despite the embed expecting Flash content.

**Is the problem with the Ruffle desktop app, extension, or self-hosted version?**
Extension (tested with latest nightly on Firefox Developer Edition)

**What platform are you using?**
Windows 10

**Comment by kmeisthax** [comment]
As far as I can tell, YouTube just fires back HTML and the browser treats the embed tag as an iframe in that case. Which means we need to check the MIME type of the response of the root movie and either not polyfill or replace the polyfill with an iframe - depending on where and how we place the check.

I'm not entirely sure how to go about this without either double-fetching the embed or undoing some of the work we did on doing root movie fetch on the Player side. I'll leave that up to whoever implements the fix; the only constraint I have on all of this is that we shouldn't expose a JavaScript API that would make it trivially easy to circumvent domain locks in games (in the DMCA 1201 sense) unless it's absolutely necessary to implement it that way.

**Comment by danielhjacobs** [comment]
@kmeisthax You're right, but I believe this is a strange consequence of the HTML standard for embeds:

https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-embed-type:

1. If another task has since been queued to run the embed element setup steps for element, then return.

2. If element has a src attribute set, then:
        1. ...
        2. ...
        3. ...
        4. Fetch request.
            1. If another task has since been queued to run the embed element setup steps for element, then return.
            2. ...
            3. Let type be the result of determining the type of content given element and response.
            4. Switch on type

[Type of content](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#concept-embed-type):
1. If element has a type attribute, **and that attribute's value is a type that a plugin supports**, then return the value of the type attribute.
2. If the path component of response's url matches a pattern that a plugin supports, then return the type that that plugin can handle.
3. If **response has explicit Content-Type metadata, and that value is a type that a plugin supports, then return that value**.

**Basically, because there is no plugin that supports Flash by default on browsers anymore, the embed defaults to the Content-Type metadata.**

**Comment by n0samu** [comment]
Given what I found in https://github.com/ruffle-rs/ruffle/pull/6520#issuecomment-1080151967, I don't think that the above theory is correct. Have you seen this behavior happen for anything besides YouTube embeds? 

My interpretation of the text you cited is that because no plugin is registered to handle the content type `text/html`, the "type of content" would be set to `null`. (You omitted step 4 in the [type of content](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#concept-embed-type) steps, which says "return null").

Now going back to the "Switch on _type_" step, it says that if the "type of content" is null, the action is to [display no plugin](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#display-no-plugin), which just means that the browser should 'display an indication that no plugin could be found for the element' instead of displaying any content. Nowhere does it say that the browser should treat the content as an iframe.
